src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -67,7 +69,17 @@ protected override void Dispose(bool disposing)
 
         public override long Seek(long offset, SeekOrigin origin)
         {
-            throw new NotSupportedException();
+            if (origin == SeekOrigin.Current)
+            {
+                offset += this.Position;
+            }
+            else if (origin == SeekOrigin.End)
+            {
+                offset += this.buffer.Capacity;
```

Comment:
should be `offset += this.buffer.WriterIndex;` as we consider WriterIndex the end of readable space